NebulaClient/MonoBehaviours/MultiplayerSession.cs

```diff
@@ -112,7 +112,72 @@ private void OnPlayerAnimationUpdate(PlayerAnimationUpdate packet)
 
         private void OnVegeMined(VegeMined packet)
 	    {
-            GameMain.localPlanet?.factory?.RemoveVegeWithComponents(packet.VegeID);
-	    }
+            PlanetData planet = null;
+
+            if(GameMain.localPlanet?.id == packet.PlanetID)
+            {
+                planet = GameMain.localPlanet;
+                if(planet == null)
+                {
+                    return;
+                }
+            }
+            else
+            {
+                planet = GameMain.galaxy?.PlanetById(packet.PlanetID);
+                if(planet == null)
+                {
+                    return;
+                }
+            }
```

Comment:
Can't you just always do `GameMain.galaxy?.PlanetById(packet.PlanetID)` even if it's your LocalPlanet ?

Reply:
I guess that should be possible.